pkg/scheduler/watchers/membercluster/controller.go

```diff
@@ -194,14 +193,8 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 			//
 			// Note that the controller runs only when label changes happen on joined clusters.
 			clusterKObj := klog.KObj(newCluster)
-			if newCluster.Spec.State == clusterv1beta1.ClusterStateJoin && !reflect.DeepEqual(oldCluster.Labels, newCluster.Labels) {
-				klog.V(2).Info("An member cluster label change has been detected", "memberCluster", clusterKObj)
-				return true
-			}
-
-			// The cluster has left.
-			if oldCluster.Spec.State == clusterv1beta1.ClusterStateJoin && newCluster.Spec.State == clusterv1beta1.ClusterStateLeave {
-				klog.V(2).Info("A member cluster has left the fleet", "memberCluster", clusterKObj)
+			if !reflect.DeepEqual(oldCluster.Labels, newCluster.Labels) {
```

Comment:
check if the deletiontimestamp is not added in the newCluster?

Reply:
one can't update a "deleting" object so it can't have deletionTimestamp.

Reply:
Deleting a resource means add a deletiontimestamp.
 https://github.com/Azure/fleet/blob/main/pkg/scheduler/watchers/clusterresourceplacement/controller.go#L75-L89 which has been validated by integration tests.